fastestimator/trace/io/model_saver.py

```diff
@@ -31,16 +33,32 @@ class ModelSaver(Trace):
         model: A model instance compiled with fe.build.
         save_dir: Folder path into which to save the `model`.
         frequency: Model saving frequency in epoch(s).
+        max_to_keep: Maximum number of latest saved files to keep. If None, all models will be saved.
     """
-    def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], save_dir: str, frequency: int = 1) -> None:
+    def __init__(self,
+                 model: Union[tf.keras.Model, torch.nn.Module],
+                 save_dir: str,
+                 frequency: int = 1,
+                 max_to_keep: Optional[int] = None) -> None:
         super().__init__(mode="train")
         self.model = model
         self.save_dir = save_dir
         self.frequency = frequency
+        self.max_to_keep = max_to_keep
```

Comment:
queue.Queue() is a lot of overhead for this, and a lot of extra variables. instead eliminate self.max_to_keep and the 'if max_to_keep' condition here and...

from collections import deque
...
self.file_queue = deque([None] * (max_to_keep or 0), maxlen = max_to_keep or 0)